### PR TITLE
18068 - Namerequest UI: Editing the Resubmit NR has the first name choice as empty.

### PR DESCRIPTION
**Pull request by Rajandeep98**
*Issue #18068 https://github.com/bcgov/entity/issues/18068

*Description of changes:*
Updated condition for applying designation for resubmission for SP/GP

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-740-qail5x3t.web.app

**Review by severinbeauvais — APPROVED** [review]
LGTM

**Review by severinbeauvais — CHANGES_REQUESTED** [review]
Removing my approval -- I didn't looks closely enough at the code. The code change will always be True because it's not comparing the NR's type with an enum; it's looking at the truthiness of the enums directly, which is always True.

Rajandeep and I looked at this together and we're not sure why this code exists -- can't the designation from the original NR simply be copied in to the new NR? This code suggests that some  NRs have a designation in the name but the designation field is empty. It would be good to have an example. We could then handle those specially.

The other option of exempting some types that we know don't have designations is less optimal but is probably safe and could be implemented as a last resort.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-740-qail5x3t.web.app

File: src/store/actions.ts
Comment (severinbeauvais) [inline src/store/actions.ts:496]: I'd name the flag something useful.

Also, you can simplify this...
```
const flag = (your boolean statement)
```

Comment (severinbeauvais) [reply src/store/actions.ts:496]: Thanks for the very useful comment!

**Review by severinbeauvais — APPROVED** [review]
I approve this solution.

**Review by eve-git — APPROVED** [review]

**Comment by severinbeauvais** [comment]
Do not merge this until after the release on Tuesday, October 31.

File: src/store/actions.ts
Comment (severinbeauvais) [reply src/store/actions.ts]: Thanks for the update. Sorry for being so picky. Your code looks good and I've approved this PR but I think you have to wait a couple of days before merging it (check with Omid).

**Review by ozamani9gh — APPROVED** [review]
looks good

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-740-qail5x3t.web.app

**Pull request merged**
